tests/visualization_tests/test_edf.py

```diff
@@ -1,106 +1,166 @@
+    _assert_target_label(figure)
 
     # Test with two studies.
     study1 = create_study(direction=direction)
     study1.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
     figure = plot_edf([study0, study1])
-    for points in figure.data:
-        _validate_edf_values(points["y"])
-    assert len(figure.data) == 2
-    figure = plot_edf((study0, study1))
-    for points in figure.data:
-        _validate_edf_values(points["y"])
-    assert len(figure.data) == 2
+    _assert_target_label(figure)
 
     # Test with a customized target value.
     study0 = create_study(direction=direction)
     study0.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
     with pytest.warns(UserWarning):
         figure = plot_edf(study0, target=lambda t: t.params["x"])
-    _validate_edf_values(figure.data[0]["y"])
-    assert len(figure.data) == 1
+    _assert_target_label(figure)
 
     # Test with a customized target name.
     study0 = create_study(direction=direction)
     study0.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
     figure = plot_edf(study0, target_name="Target Name")
-    _validate_edf_values(figure.data[0]["y"])
-    assert len(figure.data) == 1
-    assert figure.layout.xaxis.title.text == "Target Name"
+    _assert_target_label(figure, expected="Target Name")
+
+
+def test_empty_edf_info() -> None:
+
+    edf_info = _get_edf_info([])
+    assert len(edf_info.lines) == 0
+    assert len(edf_info.x_values) == 0
+
+    study = create_study()
+    edf_info = _get_edf_info(study)
+    assert len(edf_info.lines) == 0
+    assert len(edf_info.x_values) == 0
+
+    trial = study.ask()
+    study.tell(trial, state=optuna.trial.TrialState.PRUNED)
+    edf_info = _get_edf_info(study)
+    assert len(edf_info.lines) == 0
+    assert len(edf_info.x_values) == 0
 
 
 @pytest.mark.parametrize("value", [float("inf"), -float("inf"), float("nan")])
 def test_nonfinite_removed(value: int) -> None:
 
     study = prepare_study_with_trials(value_for_first_trial=value)
-    figure = plot_edf(study)
-    assert all(np.isfinite(figure.data[0]["x"]))
-
-
-@pytest.mark.parametrize(
-    "objective,value",
-    [
-        (0, float("inf")),
-        (0, -float("inf")),
-        (0, float("nan")),
-        (1, float("inf")),
-        (1, -float("inf")),
-        (1, float("nan")),
-    ],
-)
+    edf_info = _get_edf_info(study)
+
+    x_values = edf_info.x_values
+    min_objective = 1.0
+    max_objective = 2.0
+    assert np.min(x_values) == min_objective
+    assert np.max(x_values) == max_objective
+    assert len(x_values) == NUM_SAMPLES_X_AXIS
+
+    lines = edf_info.lines
+    assert len(lines) == 1
+    assert lines[0].study_name == study.study_name
+    _validate_edf_values(lines[0].y_values)
+
+
+@pytest.mark.parametrize("objective", (0, 1))
+@pytest.mark.parametrize("value", (float("inf"), -float("inf")))
 def test_nonfinite_multiobjective(objective: int, value: int) -> None:
```

Comment:
```suggestion
def test_nonfinite_multiobjective(objective: int, value: float) -> None:
```
The type annotation is inconsistent with the test cases.

Reply:
Nice catch. This came from the current codebase, so I just copied it.